my-vault/cmd/main.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	 _ "github.com/lib/pq"
+	"github.com/bmasaiti/go-projects/my-vault/internal/handlers"
+	"github.com/bmasaiti/go-projects/my-vault/internal/storage"
+)
+
+// Create a HTTP server that allows you to create, delete, read and list “secrets”.
+//  The following paths and verbs should be used for each ({} denotes a path variable):
+// Create: POST /secrets
+// Delete: DELETE /secrets/{secret_id}
+// Read: GET /secrets/{secret_id}
+// List: GET /secrets
+// Evertything can be stored in-memory and no need for authentication for now.
+// Each endpoint should accept and return a JSON formatted string (fields are up to you).
+
+// curl \
+//     --header "X-Vault-Token: ..." \
+//     --request POST \
+//     --data @payload.json \
+//     https://127.0.0.1:8200/v1/secret/data/my-secret
+
+//curl -d "@request_payload.json" -X POST  http://localhost:8000/secrets
+
+//TODO: Create a db object that can be swapped out for a  real db
+//TODO: add decent logging.
+//TODO: Proper error handling.
+//TODO: encrypting secret object.
+
+
+func getEnv(key, defaultValue string) string {
+	if value := os.Getenv(key); value != "" {
+		return value
+	}
+	return defaultValue
+}
```

Comment:
There is a new function in the Go standard library `cmp` package that makes this even easier:
```suggestion
func getEnv(key, defaultValue string) string {
	return cmp.Or(os.Getenv(key), defaultValue)
}
```
It will return the first value that is not the zero value of the type.